wordpress-mobile/WordPress-Android
# Feature/save post for later native caching

Fixes #7699 

For Phase 1 we are going to utilize native WebView cache. Simple WebView cache is a "best-effort" kind-of cache, as we can't guarantee that it will stick for a long time, or if we would be able to fully cache post media. 

WebView caching runs on UI thread, so I'm running it in a dedicated fragment (not attached to any container). Because orientation change destroys UI, caching will be restarted from scratch. 

Caching is quiet, and we do not notify users of progress or errors.
Caching is not performed when "Saved posts" filter is selected.

To test:
- Bookmark a post you haven't opened before (clear cache for a sanity check) with images in it.
- Turn off the internet and open the post - images should be in it.

Next steps:
For Phase 2 we have a couple of options:

- Utilize [HTML5 App Cache API](https://www.html5rocks.com/en/tutorials/appcache/beginner/). If we can implement this server-side this might be the easiest way to do cache inside the app, as WebView supports it natively. Alternatively, we can try to create and inject the manifest manually, but this is quite hacky.

- Utilize WebViewClient to intercept loading of resources and cache images manually. This is a bit more involved process, but we will have more control over cache and will be able to store it as app data (more persistent).
